src/metabase/query_processor/pivot.clj

```diff
@@ -173,11 +175,11 @@
                                                            [:pivot-rows {:optional true} [:maybe ::pivot-rows]]
                                                            [:pivot-cols {:optional true} [:maybe ::pivot-cols]]]]
   (try
-    (let [all-breakouts (lib/breakouts query)]
+    (let [all-breakouts  (lib/breakouts query)]
       (for [breakout-indexes (u/prog1 (breakout-combinations (count all-breakouts) pivot-rows pivot-cols)
-                                (log/tracef "Using breakout combinations: %s" (pr-str <>)))
-            :let              [group-bitmask (group-bitmask (count all-breakouts) breakout-indexes)]]
-        (-> query
+                               (log/tracef "Using breakout combinations: %s" (pr-str <>)))
+            :let             [group-bitmask (group-bitmask (count all-breakouts) breakout-indexes)]]
+        (-> (assoc-in query [:info :original-query] (str query))
```

Comment:
Here I keep the original query as a string here because when I just kept it as a data structure, the QP complained. I don't think turning it into a string (and subsequently reading it) is actually the right solution, but help or a suggestion would be appreciated here!

Reply:
yeah this is definitely not a great solution, you should just fix where the QP was complaining. What was the error?

Reply:
Various parts of the qp were operating on the query as if it were THE query. That is, some normalization/validation stuff runs recursively through all maps and tokens. 
So, when it was just a string, nothing wen wrong. But, when it's a query, it was complaining about duplicate uuids. But that was fixed with me only applying the original-query key to the first of the pivot sub queries.

A couple normalization spots were going wrong too. I think I've got a few things sorted, but I want to do a follow up PR with more tests + better understanding of the most correct way to solve things.